core/src/main/java/com/kin/ecosystem/core/util/ErrorUtil.java

```diff
@@ -39,31 +38,32 @@
 	public static final int ERROR_CODE_REQUEST_TIMEOUT = 408;
 	public static final int ERROR_CODE_CONFLICT = 409;
 	public static final int ERROR_CODE_INTERNAL_SERVER_ERROR = 500;
+	public static final int ERROR_CODE_TRANSACTION_FAILED_ERROR = 700;
 
-	public static final int ERROR_CODE_USER_NOT_FOUND = 4046;
 	public static final int ERROR_CODE_EXTERNAL_ORDER_ALREADY_COMPLETED = 4091;
 
 	public static KinEcosystemException fromApiException(ApiException apiException) {
 		KinEcosystemException exception;
 		if (apiException == null) {
-			exception = createUnknownException(null);
+			exception = createUnknownServiceException(null);
 		} else {
 			final int apiCode = apiException.getCode();
 			switch (apiCode) {
 				case ERROR_CODE_BAD_REQUEST:
 				case ERROR_CODE_UNAUTHORIZED:
 				case ERROR_CODE_NOT_FOUND:
-					if (apiException.getResponseBody() != null
-						&& apiException.getResponseBody().getCode() == ERROR_CODE_USER_NOT_FOUND) {
-						exception = new ServiceException(ServiceException.USER_NOT_FOUND,
-							USER_NOT_FOUND_ON_ECOSYSTEM_SERVER, apiException);
-						break;
-					}
 				case ERROR_CODE_CONFLICT:
 				case ERROR_CODE_INTERNAL_SERVER_ERROR:
-					exception = new ServiceException(ServiceException.SERVICE_ERROR,
-						THE_ECOSYSTEM_SERVER_RETURNED_AN_ERROR,
-						apiException);
+				case ERROR_CODE_TRANSACTION_FAILED_ERROR:
+					Error error = apiException.getResponseBody();
+					if (error != null) {
+						String msg = error.getMessage();
+						exception = new ServiceException(ServiceException.SERVICE_ERROR,
+							(msg != null && !msg.isEmpty()) ? msg : THE_ECOSYSTEM_SERVER_RETURNED_AN_ERROR,
```

Comment:
if a message is null try we can try to get apiException.getMessage()

Reply:
it is like that see  `createUnknownServiceException` method